lib/glific/templates/interactive_templates.ex

```diff
@@ -384,4 +386,313 @@ defmodule Glific.Templates.InteractiveTemplates do
+            "items" => items_translated,
+            "type" => "list"
+          }
+
+          Map.put(
+            acc,
+            Integer.to_string(Map.get(language_code_map, lang_code)),
+            translated_template
+          )
+      end
+    end)
+  end
+
+  @spec translate_location_request(map(), map(), map(), non_neg_integer()) :: map()
+  defp translate_location_request(content, active_languages, language_code_map, organization_id) do
+    content_to_translate = [content["body"]["text"]]
+
+    Enum.reduce(active_languages, %{}, fn {lang_name, lang_code}, acc ->
+      translations =
+        if lang_name == "English" do
+          {:ok, content_to_translate}
+        else
+          GoogleTranslate.translate(content_to_translate, "English", lang_name,
+            org_id: organization_id
+          )
+        end
+
+      case translations do
+        {:ok, [translated_text]} ->
+          translated_template = %{
+            "action" => content["action"],
+            "body" => %{"text" => translated_text, "type" => content["body"]["type"]},
+            "type" => content["type"]
+          }
+
+          Map.put(
+            acc,
+            Integer.to_string(Map.get(language_code_map, lang_code)),
+            translated_template
+          )
+      end
+    end)
+  end
+
+  @spec build_content_to_translate(map()) :: list
+  defp build_content_to_translate(content) do
+    [
+      content["title"],
+      content["body"]
+    ] ++
+      Enum.map(content["globalButtons"], fn button ->
+        button["title"]
+      end) ++
+      Enum.flat_map(content["items"], fn item ->
+        [
+          item["title"],
+          item["subtitle"]
+        ] ++
+          Enum.flat_map(item["options"], fn option ->
+            [option["title"], option["description"]]
+          end)
+      end)
+  end
+
+  @spec translate_global_buttons([String.t()], [map()]) :: [map()]
+  defp translate_global_buttons(global_buttons_translations, global_buttons) do
+    Enum.zip(global_buttons_translations, global_buttons)
+    |> Enum.map(fn {translated_title, button} ->
+      Map.put(button, "title", translated_title)
+    end)
+  end
+
+  @spec add_url_if_present(map(), map()) :: map()
+  defp add_url_if_present(map, content) do
+    if Map.has_key?(content["content"], "url") do
+      Map.put(map, "url", content["content"]["url"])
+    else
+      map
+    end
+  end
+
+  @spec do_options_translated(map(), list()) :: list()
+  defp do_options_translated(content, options) do
+    Enum.zip(Enum.map(content["options"], fn option -> option["type"] end), options)
+    |> Enum.map(fn {type, title} -> %{"type" => type, "title" => title} end)
+  end
+
+  @spec do_items_translated(map(), list()) :: list()
+  defp do_items_translated(content, items_translations) do
+    options_length = length(Enum.at(content["items"], 0)["options"])
+
+    Enum.chunk_every(items_translations, 2 + options_length * 2)
```

Comment:
`2 + options_length * 2` what does this do?

Reply:
`"1" => %{
    "body" => "Glific",
    "globalButtons" => [%{"title" => "button text", "type" => "text"}],
    "items" => [
      %{
        "options" => [
          %{
            "description" => "Flow Editor for creating flows",
            "title" => "Custom Flows",
            "type" => "text"
          },
          %{
            "description" => "DataStudio for report generation",
            "title" => "Analytic Reports",
            "type" => "text"
          },
          %{
            "description" => "Dialogflow for AI/ML",
            "title" => "ML/AI",
            "type" => "text"
          }
        ],
        "subtitle" => "Glific Features",
        "title" => "Glific Features"
      },
      %{
        "options" => [
          %{
            "description" => "Sharing education content with school student",
            "title" => "Educational programs",
            "type" => "text"
          }
        ],
        "subtitle" => "Glific Usecases",
        "title" => "Glific Usecases"
      }
    ],
    "title" => "Interactive list",
    "type" => "list"
  }`
  
this is the JSON. Here, the option length is the number of elements in the options list, which is 3 in the above case. We multiply this by 2 because we are translating only the title and description, another 2 that we are adding is for the subtitle and title.